src/app/Helpers/Stats.php

```diff
@@ -23,12 +23,100 @@ public static function countFutureAvailability() {
       return Location::available()->count();
     }
 
+    /**
+     * Get the number of locations that have never had their information
+     * updated.
+     */
+    public static function countLocationsNeverUpdated() {
+        return count(DB::select(DB::raw("select id from locations where id NOT IN (select location_id from availabilities group by location_id)")));
+    }
+
+    /**
+     * Get the number of providers whose most recent updates occurred longer
+     * than $days days ago.
+     *
+     * NB: This count will not include sites that have never been updated.
+     *
+     * @param days The cutoff from which to measure whether sites
+     * have been updated.
+     */
+    public static function countLocationsUpdatedLongerAgoThan(int $days) {
+        $results = Availability::where(DB::raw("floor((extract(epoch from NOW() - updated_at))/86400)"), ">", $days)
+          ->distinct()
+          ->count('location_id');
+        return $results;
+    }
+
+    /**
+     * Build an array of counts, where each index's count represents the number
+     * of providers updated at _index_ days in the past. For example, the array
+     * of counts that looks like
+     * 5, 6, 9, 1, 3
+     * means that 5 sites were last updated today, 6 were last updated
+     * yesterday, 9 where updated 2 days ago, 1 was updated 3 days ago and 3
+     * were updated 4 days ago.
+     *
+     * @param forDays The number of days that the histogram will contain.
+     */
+    public static function lastUpdatedHistogramData(int $forDays = 30) {
+        $buckets = [];
+        /**
+         * Count the number of sites updated during each of the last
+         * $forDays days.
+         */
+        $results = DB::select(DB::raw("select floor((extract(epoch from NOW() - updated_at))/86400) as since_update, count(*) from (select max(updated_at) as updated_at from availabilities group by location_id) as location_update_times where floor((extract(epoch from NOW() - updated_at))/86400) < ? group by since_update order by since_update asc"), [$forDays]);
```

Comment:
You might add some linebreaks so that the SQL is easier to read. Whitespace shouldn't affect the result.